### DISCUSSION TITLE
Delayed in entering PIN as String (Mobile Application)

### DISCUSSION THREAD

**Discussion by ayushsinghcool** [appium/appium#16762] [Q&A]
While automating the flow I was stuck in slowness of entering user PIN for login. In Earlier version of java client 6.0.0 it is working fine but I have updated the javaclient to 8.X and appium 2.X, I am facing this issue.

I tried every possible way but not got success. This issue only exists for USER PIN Only. 

Code for refrence 
XML PAGE SOURCE

`<?xml version="1.0" encoding="UTF-8"?>
<hierarchy index="0" class="hierarchy" rotation="0" width="1440" height="2560">
                                    **### <android.widget.EditText index="0" package="ma.inwi.money.alpha" class="android.widget.EditText" text="" resource-id="ma.inwi.money.alpha:id/" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="true" scrollable="false" selected="false" bounds="[152,1429][275,1611]" displayed="true" />
                                    <android.widget.EditText index="1" package="ma.inwi.money.alpha" class="android.widget.EditText" text="" resource-id="ma.inwi.money.alpha:id/" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="true" scrollable="false" selected="false" bounds="[320,1429][443,1611]" displayed="true" />
                                    <android.widget.EditText index="2" package="ma.inwi.money.alpha" class="android.widget.EditText" text="" resource-id="ma.inwi.money.alpha:id/" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="true" scrollable="false" selected="false" bounds="[488,1429][611,1611]" displayed="true" />
                                    <android.widget.EditText index="3" package="ma.inwi.money.alpha" class="android.widget.EditText" text="" resource-id="ma.inwi.money.alpha:id/" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" long-clickable="false" password="true" scrollable="false" selected="false" bounds="[656,1429][779,1611]" displayed="true" />**
</hierarchy>`

Java Code

` @AndroidFindBy( xpath = "//*[@class='android.widget.EditText' and ./parent::*[@class='android.widget.RelativeLayout']]")
    private List<WebElement> password;

or

 @AndroidFindBy(xpath = "//android.widget.EditText[@password = 'true']")
    private WebElement password;

 public InwiLoginPageObjects setData( String pin){
       int i=0;
        for(WebElement pass : password){
            setText(pass,""+pin.charAt(i),"Entering PIN");
            i++;
        }
        return this;
    }
`

LOGS

`[HTTP] 
[HTTP] --> POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/elements
[HTTP] {"using":"xpath","value":"//*[@class='android.widget.EditText' and ./parent::*[@class='android.widget.RelativeLayout']]"}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Calling AppiumDriver.findElements() with args: ["xpath","//*[@class='android.widget.EditText' and ./parent::*[@class='android.widget.RelativeLayout']]","36b952f2-7d1a-49d4-912c-caf8d3951c20"]
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Valid locator strategies for this request: xpath, id, class name, accessibility id, css selector, -android uiautomator
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Waiting up to 0 ms for condition
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Matched '/elements' to command name 'findElements'
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Proxying [POST /elements] to [POST http://127.0.0.1:8201/session/a14d74b2-01a7-498b-986c-14863d1ba943/elements] with body: {"strategy":"xpath","selector":"//*[@class='android.widget.EditText' and ./parent::*[@class='android.widget.RelativeLayout']]","context":"","multiple":true}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Got response with status 200: {"sessionId":"a14d74b2-01a7-498b-986c-14863d1ba943","value":[{"ELEMENT":"00000000-0000-018d-ffff-ffff00000036","element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000036"},{"ELEMENT":"00000000-0000-018d-ffff-ffff00000035","element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000035"},{"ELEMENT":"00000000-0000-018d-ffff-ffff00000034","element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000034"},{"ELEMENT":"00000000-0000-018d-ffff-ffff00000033","element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000033"}]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Responding to client with driver.findElements() result: [{"element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000036","ELEMENT":"00000000-0000-018d-ffff-ffff00000036"},{"element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000035","ELEMENT":"00000000-0000-018d-ffff-ffff00000035"},{"element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000034","ELEMENT":"00000000-0000-018d-ffff-ffff00000034"},{"element-6066-11e4-a52e-4f735466cecf":"00000000-0000-018d-ffff-ffff00000033","ELEMENT":"00000000-0000-018d-ffff-ffff00000033"}]
[HTTP] <-- POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/elements 200 10346 ms - 523
[HTTP] 
[HTTP] --> POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000036/value
[HTTP] {"id":"00000000-0000-018d-ffff-ffff00000036","text":"1","value":["1"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Calling AppiumDriver.setValue() with args: [["1"],"00000000-0000-018d-ffff-ffff00000036","36b952f2-7d1a-49d4-912c-caf8d3951c20"]
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Matched '/element/00000000-0000-018d-ffff-ffff00000036/value' to command name 'setValue'
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Added 'value' property ["1"] to 'setValue' request body
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Proxying [POST /element/00000000-0000-018d-ffff-ffff00000036/value] to [POST http://127.0.0.1:8201/session/a14d74b2-01a7-498b-986c-14863d1ba943/element/00000000-0000-018d-ffff-ffff00000036/value] with body: {"elementId":"00000000-0000-018d-ffff-ffff00000036","text":"1","replace":false,"value":["1"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Got response with status 200: {"sessionId":"a14d74b2-01a7-498b-986c-14863d1ba943","value":null}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Responding to client with driver.setValue() result: null
[HTTP] <-- POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000036/value 200 40181 ms - 14
[HTTP] 
[HTTP] --> POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000035/value
[HTTP] {"id":"00000000-0000-018d-ffff-ffff00000035","text":"2","value":["2"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Calling AppiumDriver.setValue() with args: [["2"],"00000000-0000-018d-ffff-ffff00000035","36b952f2-7d1a-49d4-912c-caf8d3951c20"]
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Matched '/element/00000000-0000-018d-ffff-ffff00000035/value' to command name 'setValue'
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Added 'value' property ["2"] to 'setValue' request body
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Proxying [POST /element/00000000-0000-018d-ffff-ffff00000035/value] to [POST http://127.0.0.1:8201/session/a14d74b2-01a7-498b-986c-14863d1ba943/element/00000000-0000-018d-ffff-ffff00000035/value] with body: {"elementId":"00000000-0000-018d-ffff-ffff00000035","text":"2","replace":false,"value":["2"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Got response with status 200: {"sessionId":"a14d74b2-01a7-498b-986c-14863d1ba943","value":null}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Responding to client with driver.setValue() result: null
[HTTP] <-- POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000035/value 200 40195 ms - 14
[HTTP] 
[HTTP] --> POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000034/value
[HTTP] {"id":"00000000-0000-018d-ffff-ffff00000034","text":"3","value":["3"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Calling AppiumDriver.setValue() with args: [["3"],"00000000-0000-018d-ffff-ffff00000034","36b952f2-7d1a-49d4-912c-caf8d3951c20"]
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Matched '/element/00000000-0000-018d-ffff-ffff00000034/value' to command name 'setValue'
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Added 'value' property ["3"] to 'setValue' request body
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Proxying [POST /element/00000000-0000-018d-ffff-ffff00000034/value] to [POST http://127.0.0.1:8201/session/a14d74b2-01a7-498b-986c-14863d1ba943/element/00000000-0000-018d-ffff-ffff00000034/value] with body: {"elementId":"00000000-0000-018d-ffff-ffff00000034","text":"3","replace":false,"value":["3"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Got response with status 200: {"sessionId":"a14d74b2-01a7-498b-986c-14863d1ba943","value":null}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Responding to client with driver.setValue() result: null
[HTTP] <-- POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000034/value 200 40178 ms - 14
[HTTP] 
[HTTP] --> POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000033/value
[HTTP] {"id":"00000000-0000-018d-ffff-ffff00000033","text":"4","value":["4"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Calling AppiumDriver.setValue() with args: [["4"],"00000000-0000-018d-ffff-ffff00000033","36b952f2-7d1a-49d4-912c-caf8d3951c20"]
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Matched '/element/00000000-0000-018d-ffff-ffff00000033/value' to command name 'setValue'
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Added 'value' property ["4"] to 'setValue' request body
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Proxying [POST /element/00000000-0000-018d-ffff-ffff00000033/value] to [POST http://127.0.0.1:8201/session/a14d74b2-01a7-498b-986c-14863d1ba943/element/00000000-0000-018d-ffff-ffff00000033/value] with body: {"elementId":"00000000-0000-018d-ffff-ffff00000033","text":"4","replace":false,"value":["4"]}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Got response with status 200: {"sessionId":"a14d74b2-01a7-498b-986c-14863d1ba943","value":null}
[debug] [AndroidUiautomator2Driver@e61e (36b952f2)] Responding to client with driver.setValue() result: null
[HTTP] <-- POST /session/36b952f2-7d1a-49d4-912c-caf8d3951c20/element/00000000-0000-018d-ffff-ffff00000033/value 200 40106 ms - 14
[HTTP] `

**Comment by KazuCocoa** [comment]
What version of Appium did you use in Java 6?
Could you share the full appium log with timestamp as GIST?
